security/azure/restrict_rdp_internet/azure_restrict_rdp_inet.pt

```diff
@@ -0,0 +1,203 @@
+rs_pt_ver 20180301
+type "policy"
+short_description "A policy that sends email notifications when an Azure Network Security Group has RDP open to the internet. See the [README](https://github.com/flexera/policy_templates/tree/master/security/azure/restrict_rdp_internet) and [docs.flexera.com/flexera/EN/Automation](https://docs.flexera.com/flexera/EN/Automation/AutomationGS.htm) to learn more."
+long_description ""
+severity "high"
+category "Security"
+default_frequency "hourly"
+info(
+  version: "2.0",
+  provider: "Azure",
+  service: "Network Security Group",
+  policy_set: "CIS",
+  cce_id: "",
+  benchmark_control: "6.1",
+  benchmark_version: "1.4.0",
+  cis_controls: "[\"9.2v7\"]",
+  nist: "SC-8"
+)
+
+###############################################################################
+# Parameters
+###############################################################################
+
+parameter "param_email" do
+  type "list"
+  label "Email addresses of the recipients you wish to notify"
+end
+
+parameter "param_azure_endpoint" do
+  type "string"
+  label "Azure Endpoint"
+  allowed_values "management.azure.com", "management.chinacloudapi.cn"
+  default "management.azure.com"
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+credentials "azure_auth" do
+  schemes "oauth2"
+  label "Azure"
+  description "Select the Azure Resource Manager Credential from the list."
+  tags "provider=azure_rm"
+end
+
+###############################################################################
+# Pagination
+###############################################################################
+
+pagination "azure_pagination" do
+  get_page_marker do
+    body_path "nextLink"
+  end
+  set_page_marker do
+    uri true
+  end
+end
+
+
+###############################################################################
+# Datasources
+###############################################################################
+
+datasource "ds_subscriptions" do
+  request do
+    auth $azure_auth
+    pagination $azure_pagination
+    host $param_azure_endpoint
+    path "/subscriptions/"
+    query "api-version","2019-06-01"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "subscriptionId", jmes_path(col_item,"subscriptionId")
+      field "displayName", jmes_path(col_item,"displayName")
+      field "state", jmes_path(col_item,"state")
+    end
+  end
+end
+
+datasource "ds_azure_resources" do
+  iterate $ds_subscriptions
+  request do
+    auth $azure_auth
+    pagination $azure_pagination
+    host $param_azure_endpoint
+    path join(["/subscriptions/", val(iter_item,"subscriptionId"),"/providers/Microsoft.Network/networkSecurityGroups"])
+    query "api-version","2021-05-01"
+    header "User-Agent", "RS Policies"
+    ignore_status [404]
```

Comment:
Is this needed for the policy to function? We generally don't want to ignore error statuses except in specific cases where the error doesn't actually indicate a problem.

Reply:
If you are referring to "ignore_status [404]" most likely not. I will remove and retest.

Reply:
Tested fine without. Removed.